plugins/trace_api_plugin/store_provider.cpp

```diff
@@ -118,16 +122,32 @@ namespace eosio::trace_api {
          while (offset < end) {
             yield();
             fc::raw::unpack(ds, entry);
-            FC_ASSERT( std::holds_alternative<block_trxs_entry>(entry) == true, "unpacked data should be a block trxs entry" );
-            const auto& trxs_entry = std::get<block_trxs_entry>(entry);
-            for (auto i = 0U; i < trxs_entry.ids.size(); ++i){
-               if (trxs_entry.ids[i] == trx_id)
-                  return trxs_entry.block_num;
+            if (std::holds_alternative<block_trxs_entry>(entry)) {
+               const auto& trxs_entry = std::get<block_trxs_entry>(entry);
+               for (auto i = 0U; i < trxs_entry.ids.size(); ++i) {
+                  if (trxs_entry.ids[i] == trx_id) {
+                     trx_entries++;
+                     trx_block_num = trxs_entry.block_num;
+                  }
+               }
+            } else if (std::holds_alternative<lib_entry_v0>(entry)) {
+               auto lib = std::get<lib_entry_v0>(entry).lib;
+               if (trx_entries > 0 && lib >= trx_block_num) {
+                  return trx_block_num;
+               }
+            } else {
+               FC_ASSERT( false, "unpacked data should be a block trxs entry or a lib entry" );;
             }
             offset = trx_id_file.tellp();
          }
          slice_number++;
       }
+
+      // no irreversible blocks after the block containing the target trx, e.g., it happens
+      // in some integration tests that call waitForTransInBlock during bios bootstrap.
```

Comment:
Change this comment (or remove it) to something more like:

`// transaction's block is not irreversible`

Reply:
Fixed.